R/sf.R

```diff
@@ -209,18 +214,17 @@ CoordSf <- ggproto("CoordSf", CoordCartesian,
       return(data)
 
     lapply(data, function(layer_data) {
-      if (is.null(layer_data$geometry)) {
+      if (! inherits(layer_data, "sf")) {
         return(layer_data)
       }
 
-      layer_data$geometry <- sf::st_transform(layer_data$geometry, params$crs)
-      layer_data
+      sf::st_transform(layer_data, params$crs)
     })
   },
 
   transform = function(self, data, panel_params) {
-    data$geometry <- sf_rescale01(
-      data$geometry,
+    data[[ which(sapply(data, function(x) inherits(x, "sfc")))[1] ]] <- sf_rescale01(
```

Comment:
Maybe `geom_column()` should return an index so you can use it on the LHS and RHS here?